linux_os/guide/services/ssh/ssh_server/sshd_use_approved_ciphers_ordered_stig/rule.yml

```diff
@@ -71,3 +71,8 @@ warnings:
         party review by an accredited lab. While open source software is
         capable of meeting this, it does not meet FIPS-140 unless the vendor
         submits to this process.
+    {{% if 'ubuntu' in product %}}
+    - audit: |-
+        The audit returns <tt>unknown</tt> instead of <tt>pass</tt> if sshd is correctly configured
```

Comment:
This is really suspicious. The rule should return **fail** if the OS isn't FIPS certified. Returning `unknown` is unexpected in this situation. The problem is that the check `installed_OS_is_FIPS_certified` doesn't work properly on Ubuntu. The reason is that the check contains `rpminfo` OVAL objects which can't be evaluated by the version of `oscap` shipped in Ubuntu because Ubuntu doesn't provide RPM and uses APT as a packaging system instead. All the checks for installed RHEL, SLE are evaluated as `unknown`, these results propagate up and influences the overall result. This needs to be addressed in the OVAL definition for the  `installed_OS_is_FIPS_certified`. I think that build-time Jinja conditions can be added there to exclude `extend_definition` elements pointing to definitions that are describing RPM-based systems from the built Ubuntu content.

Reply:
The check in `installed_OS_is_FIPS_certified` works fine in Ubuntu, as you will see we have specific entries for FIPS certified environments: e.g. `installed_OS_is_ubuntu2004`. The problem is that ubuntu2204 is not yet certified per se but we already have DISA STIG which requires FIPS certification. So we are in the middle of a grey area here where if we add the `installed_OS_is_ubuntu2204` it wouldn't be accurate, while if we don't add it it returns `unknown` as none of the checks will pass.

Reply:
Hi @jan-cerny , thanks for pointing this out. I falsely assumed that `unknown` is the expected result of the `installed_OS_is_FIPS_certified` when the OS is not defined in the list of FIPS-certified. The "functionality" seemed convenient for our case, allowing us to differentiate between a fully compliant system (`pass`) and a system without FIPS-certification, but a compliant sshd configuration (`unknown`). But since the reason is, as you pointed out, a bug in the evaluation, I'll close the PR and we can forget this ever happened :)

That said, I was hoping you could help me understand why the requirement for FIPS certification is hard-coded into CaC, i.e. why the OS must be added to the list in `installed_OS_is_FIPS_certified`? Further, is it sensible that the requirement is OS-based and not module-based?

Reply:
@dodys I have verified on Ubuntu 22.04 that it doesn't work. As I explained above, the reason is that the check contains extend_definitions that points to definitions that are using the RPM-based OVAL tests for which the probes aren't available on Ubuntu. This problem could be addressed by the solution I outlined.

@mpurg I will try to explain my understanding of it. 

There are 2 different things:
1. the software is configured to use crypto algorithms specified in FIPS-140
2. the software is compliant with FIPS-140

In past we faced a problem that this rule passed when (1) was true but customers thought that it means (2). The problem is that unless the vendor has submitted to the certification process and software has received the certification, the software isn't compliant with FIPS-140. Therefore, we extended the check in a way that the rule will always fail on systems that haven't received the certification.

I think you have a good point regarding the modules. I think it would make sense to check for specific versions and crypto libraries instead of the OS because the certification is for specific modules, not for OS as whole.